llvm/lib/Target/AMDGPU/AMDGPUSwLowerLDS.cpp

```diff
@@ -1226,6 +1228,8 @@ bool AMDGPUSwLowerLDS::run() {
       auto &K = FuncLDSAccessInfo.NonKernelToLDSAccessMap;
       if (K.find(Func) != K.end())
         continue;
+      if (Func->isDeclaration())
+        continue;
```

Comment:
Do this first before the map lookup

Reply:
Fixed this in latest commit. Skipped adding function to map if its a declaration.